File: sdk/python/tempmail_sdk/providers/mingyukeji_lol.py
"""
mailmomy 域名变体渠道：mingyukeji.lol
固定使用域名 mingyukeji.lol 生成邮箱，收信复用 mailmomy 的 get_emails。
"""

from typing import List

from ..types import Email, EmailInfo
from . import mailmomy

CHANNEL = "mingyukeji-lol"
DOMAIN = "mingyukeji.lol"


def generate_email() -> EmailInfo:
    """生成 mingyukeji.lol 域名的临时邮箱；Token 即邮箱地址本身"""
    email = f"{mailmomy._random_local(10)}@{DOMAIN}"
    return EmailInfo(channel=CHANNEL, email=email, _token=email)


def get_emails(email: str) -> List[Email]:
    """收信直接委托给 mailmomy（同后端 API）"""
    return mailmomy.get_emails(email)

File: sdk/rust/src/providers/mingyukeji_lol.rs
/*!
 * mailmomy 域名变体 mingyukeji.lol
 * 独立自研后端，收信复用 mailmomy provider 的读信逻辑。
 */

use crate::providers::mailmomy;
use crate::types::{Channel, Email, EmailInfo};
use rand::Rng;

const DOMAIN: &str = "mingyukeji.lol";

/// 生成随机本地部分（小写字母 + 数字）
fn random_local(len: usize) -> String {
    const CHARS: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789";
    let mut rng = rand::thread_rng();
    (0..len)
        .map(|_| CHARS[rng.gen_range(0..CHARS.len())] as char)
        .collect()
}

/// 创建 mingyukeji.lol 临时邮箱（免注册，token 即邮箱地址本身）
pub fn generate_email() -> Result<EmailInfo, String> {
    let email = format!("{}@{}", random_local(10), DOMAIN);
    Ok(EmailInfo {
        channel: Channel::MingyukejiLol,
        email: email.clone(),
        token: Some(email),
        expires_at: None,
        created_at: None,
    })
}

/// 获取 mingyukeji.lol 邮件列表（复用 mailmomy 读信逻辑）
pub fn get_emails(email: &str) -> Result<Vec<Email>, String> {
    mailmomy::get_emails(email)
}
